prophyc/__init__.py

```diff
@@ -31,41 +36,61 @@ def main():
     if not opts.input_files:
         Emit.error("missing input file")
 
-    parser = get_parser(opts)
     serializers = get_serializers(opts)
     patcher = get_patcher(opts)
 
     if not serializers:
         Emit.error("missing output directives")
 
+    supplementary_nodes = []
+    if opts.isar_includes:
+        if opts.isar:
+            Emit.error('Bad usage. Isar include is intended to supplement code '
+                       'of different language.\nPass it all together as input files.')
+        if not opts.sack:
+            Emit.error('Isar defines inclusion is supported only in "sack" compilation mode.')
+
+        isar_parser = get_isar_parser()
+
+        def supplementary_parser(*parse_args):
+            return parse_content(isar_parser, patcher, *parse_args)
+
+        supple_file_parser = FileProcessor(supplementary_parser, opts.include_dirs)
+
+        for input_file in opts.isar_includes:
+            with parse_error_wrapper():
+                include_nodes = supple_file_parser(input_file)
+            basename = get_basename(input_file)
+            supplementary_nodes.append(model.Include(basename, include_nodes))
+
+        for include_name, include_nodes in get_supple_includes(supplementary_nodes):
+            generate_target_files(serializers, include_name, include_nodes)
+
+    parser = get_target_parser(opts, supplementary_nodes)
+
     def content_parser(*parse_args):
         return parse_content(parser, patcher, *parse_args)
 
     file_parser = FileProcessor(content_parser, opts.include_dirs)
 
     for input_file in opts.input_files:
-        try:
+        with parse_error_wrapper():
```

Comment:
This error guard is a good idea. Easier to read than ugly try/except crutch. Consider naming it according to its function, not its identity - e.g. `with exit_on_error():` could be more informative.